bolt/include/bolt/Profile/YAMLProfileReader.h

```diff
@@ -16,6 +16,37 @@
 namespace llvm {
 namespace bolt {
 
+/// A class for matching binary functions in functions in the YAML profile.
```

Comment:
I think it would be useful to add a general description of the algorithm, in particular if you have a reference to the paper, somewhere.